R/translate-sql-conditional.R

```diff
@@ -1,3 +1,87 @@
+sql_case_match <- function(.x, ..., .default = NULL, .ptype = NULL, error_call = current_call()) {
+  check_not_supplied(.ptype, call = error_call)
+
+  x_expr <- enexpr(.x)
+  if (!is_symbol(x_expr)) {
+    msg <- "{.arg .x} must be a variable, not a {.obj_type_friendly {.x}}."
+    cli_abort(msg, call = error_call)
+  }
+
+  formulas <- list2(...)
+  formulas <- purrr::compact(formulas)
+
+  n <- length(formulas)
+
+  if (n == 0) {
+    cli_abort("No cases provided", call = error_call)
+  }
+
+  con <- sql_current_con()
+  query <- vector("list", n)
+  value <- vector("list", n)
+
+  for (i in seq_len(n)) {
```

Comment:
Maybe this could be extracted out into `sql_case_match_clause()`? That might making testing a bit more focussed, since most of the challenge is handling the individual clauses, not assembling the whole expression.